llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -1900,6 +1900,13 @@ SDValue SelectionDAGBuilder::getValueImpl(const Value *V) {
                          DAG.getConstant(0, getCurSDLoc(), MVT::nxv16i1));
     }
 
+    if (VT.isRISCVVectorTuple()) {
+      assert(C->isNullValue() && "Can only zero this target type!");
+      return NodeMap[V] = DAG.getNode(
+                 ISD::SPLAT_VECTOR, getCurSDLoc(), VT,
+                 DAG.getConstant(0, getCurSDLoc(), MVT::getIntegerVT(8)));
```

Comment:
I'm not familiar with the RVV tuple types, but is it possible to create a SPLAT_VECTOR with a wide vector type and then perform a bitcast if that's easier?

Reply:
RVV tuple type encode `NF` information, so if we just create a normal vector type `SPLAT_VECTOR`, we are missing the information and doesn't know how to perform bitcast?

Reply:
You can create a normal SPLAT_VECTOR and bitcast it to the tuple type. In RISCVISelDAGToDAG:select, you can detect the (tupletype (bitcast (splat_vector 0))) and emit the correct number of instructions.

Reply:
Bitcast need both side of type size are the same, is it possible to cast a scalable vector type to a non-scalable type?

Reply:
Isn't the tuple type the same size as some scalable vector type?

Reply:
But its `TypeSize` is scalable, even it's same size, it's not the same.

Reply:
> But its `TypeSize` is scalable, even it's same size, it's not the same.

The tuple TypeSize should also be scalable since its size depends on vscale.